Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -387,9 +394,158 @@ namespace Tensile
                     m_fanValues[i] += newValue;
             }
 
+            rsmi_gpu_metrics_t gpuMetrics;
+            auto status = rsmi_dev_gpu_metrics_info_get(m_smiDeviceIndex, &gpuMetrics);
+            if(status != RSMI_STATUS_SUCCESS)
+            {
+                m_hasInvalidGpuMetricStatus = true;
+            }
+            else
+            {
+                if(!m_hasInvalidGpuMetricStatus)
+                {
+                    m_freqValues.push_back(gpuMetrics.average_gfxclk_frequency);
+                    m_powerValues.push_back(gpuMetrics.average_socket_power);
+                    m_tempHotspotValues.push_back(gpuMetrics.temperature_hotspot);
+                }
+            }
             m_dataPoints++;
         }
 
+        double HardwareMonitor::getAverageMedianGfxFreqPowerTemperature(int freqPowerTempIndex)
```

Comment:
Can we unify the ave/median calculation for all 3 types?
We just need to pass the reference of vector.